### ISSUE TITLE
Add `FromField` instance for `UUID`

### ISSUE THREAD

**Issue by vrom911** [nurpax/sqlite-simple#62]
`postgresql-simple` contains [such instance](https://hackage.haskell.org/package/postgresql-simple-0.5.3.0/docs/Database-PostgreSQL-Simple-FromField.html#t:FromField). It would be really helpful to have it here as well.

**Comment by sigrlami** [comment]
@vrom911 thanks for reporting. I'll take a look into it.

**Comment by kmicklas** [comment]
Postgres supports UUIDs as a native type, so there is a canonical encoding. SQLite doesn't, so we'd have to decide whether to encode them via the text representation of more compactly as binary blobs. Neither way is going to be best for all use cases.

**Comment by danwdart** [comment]
I've been using this locally, and I can make it a "pivot" package with only instances, if there's interest:

```haskell
instance FromField UUID where
    fromField f@(Field (SQLText txt) _) = case fromText txt of
        Just r  -> Ok r
        Nothing -> returnError ConversionFailed f "need a uuid"
    fromField f = returnError ConversionFailed f "need a uuid"

instance ToField UUID where
    toField = SQLText . toText
```

**Comment by malteneuss** [comment]
As there are basically two valuable encodings (as binary `BLOB` for performance, and `TEXT` for human readability in e.g. CLI),  would it be possible to add both encodings with  *`BLOB` as the default* to be fast by default like Postgres and popular Rust libraries like rusqlite do, and a newtype wrapper like `TextUUID` for textual representations?

**Comment by jchia** [comment]
Unlike Postgresql, sqlite has no UUID type. I'm skeptical about favoring the binary representation over the textual representation or vice versa, so if some change is to be made, I think it makes more sense to have 2 newtype wrappers over UUID.

**Comment by malteneuss** [comment]
True, there's no native Sqlite type, but UUID has canonical representations in BLOB or TEXT form (as described in https://www.rfc-editor.org/rfc/rfc9562.html#section-4), which are commonly used with Sqlite , see e.g. https://docs.rs/sqlx/latest/sqlx/sqlite/types/index.html#uuid.
- BLOB has the straight-forward Big-endian Hex-encoding "Saving UUIDs to binary format is done by sequencing all fields in big-endian format"
- TEXT has the common hyphenated format. "The formal definition of the UUID string representation is provided by the following ABNF .... f81d4fae-7dec-11d0-a765-00a0c91e6bf6"

For convenience it would be great to make one of those encodings the default one for the `UUID` type. I would prefer to make the more performant one (BLOB) the default to have better speeds by default for those that don't care, but either one is fine.

**Comment by malteneuss** [comment]
> I've been using this locally, and I can make it a "pivot" package with only instances, if there's interest:
> 
> instance FromField UUID where
>     fromField f@(Field (SQLText txt) _) = case fromText txt of
>         Just r  -> Ok r
>         Nothing -> returnError ConversionFailed f "need a uuid"
>     fromField f = returnError ConversionFailed f "need a uuid"
> 
> instance ToField UUID where
>     toField = SQLText . toText
> 
> Or I can pluck the code from [#89](https://github.com/nurpax/sqlite-simple/pull/89)...

Pattern matching doesn't seem to work anymore on `Field` but needs `fieldData`. So here's some ByteString UUID code to get going

```
import Data.ByteString (ByteString)
import Data.ByteString.Lazy qualified as LBS
import Data.UUID (UUID)
import Data.UUID qualified as UUID
import Database.SQLite.Simple.FromField
import Database.SQLite.Simple.ToField



-- store uuids as binary
instance ToField UUID where
  toField = SQLBlob . LBS.toStrict . UUID.toByteString

instance FromField UUID where
  fromField f = case fieldData f of
    SQLBlob blob -> case UUID.fromByteString (LBS.fromStrict blob) of
      Just uuid -> pure uuid
      Nothing -> returnError ConversionFailed f "need valid UUID"
    _ -> returnError ConversionFailed f "need a uuid BLOB"
```
